lib/src/Base/Stat/Isotropic.cxx

```diff
@@ -0,0 +1,167 @@
+  }
+  inputDimension_ = inputDimension;
+}
+
+/* 1D covariance kernel accessor */
+void Isotropic::setKernel(const CovarianceModel & oneDimensional)
+{
+  // Check if the given stationary covariance kernel has dimension one.
+  const UnsignedInteger kernelInputDimension = oneDimensional.getInputDimension();
+  if (kernelInputDimension != 1)
+  {
+    throw InvalidArgumentException(HERE) << "In Isotropic::setKernel, the supplied covariance kernel must have input dimension 1, not " << kernelInputDimension <<".";
+  }
+  kernel_ = oneDimensional;
+}
+
+const CovarianceModel & Isotropic::getKernel() const
+{
+  return kernel_;
+}
+
+/* Virtual constructor */
+Isotropic * Isotropic::clone() const
+{
+  return new Isotropic(*this);
+}
+
+/* Computation of the covariance function */
+Scalar Isotropic::computeStandardRepresentative(const Point & tau) const
+{
+  if (tau.getDimension() != inputDimension_) throw InvalidArgumentException(HERE) << "Error: expected a shift of dimension=" << inputDimension_ << ", got dimension=" << tau.getDimension();
+  const Point tauNorm = Point(1, tau.norm());
+  return kernel_.computeStandardRepresentative(tauNorm, Point(1));
+}
+
+Scalar Isotropic::computeStandardRepresentative(const Collection<Scalar>::const_iterator & s_begin,
+    const Collection<Scalar>::const_iterator & t_begin) const
+{
+  Scalar tauNormSquare = 0;
+  Collection<Scalar>::const_iterator s_it = s_begin;
+  Collection<Scalar>::const_iterator t_it = t_begin;
+  for (UnsignedInteger i = 0; i < inputDimension_; ++i, ++s_it, ++t_it)
+  {
+    const Scalar dx = *s_it - *t_it;
+    tauNormSquare += dx * dx;
+  }
+  const Point tauNorm = Point(1, std::sqrt(tauNormSquare));
+  return kernel_.computeStandardRepresentative(tauNorm, Point(1));
+}
+
+/* Gradient */
+Matrix Isotropic::partialGradient(const Point & s,
+    const Point & t) const
+{
+  if (s.getDimension() != inputDimension_) throw InvalidArgumentException(HERE) << "Error: the point s has dimension=" << s.getDimension() << ", expected dimension=" << inputDimension_;
+  if (t.getDimension() != inputDimension_) throw InvalidArgumentException(HERE) << "Error: the point t has dimension=" << t.getDimension() << ", expected dimension=" << inputDimension_;
+  const Point tau = s - t;
+  const Scalar tauNorm = tau.norm();
+  const Point normGradient(tau / tauNorm);
+  const Matrix kernelPartialGradient = kernel_.partialGradient(Point(1, tauNorm), Point(1));
+  return Matrix(inputDimension_, 1, normGradient) * kernelPartialGradient(0, 0);
+}
+
+/* Parameters accessor */
+void Isotropic::setFullParameter(const Point & parameter)
+{
+  kernel_.setFullParameter(parameter);
+}
+
+Point Isotropic::getFullParameter() const
+{
+  return kernel_.getFullParameter();
+}
+
+Description Isotropic::getFullParameterDescription() const
+{
+  return kernel_.getFullParameterDescription();
+}
+
+void Isotropic::setScale(const Point & scale)
```

Comment:
In case of error, an exception should be raised by Isotropic, not by the underlying covariance.